### PR TITLE
⬆️ (project) upgrade to python 3.9

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Recent Ansible releases require to upgrade base python release to 3.8+.

## Proposal

- [x] upgrade base python image to 3.9
- [x] unpin outdated secondary dependencies

### CODE REVIEW

File: requirements/production.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Core
 ansible==4.2.0
-jmespath==0.10.0
+#jmespath==0.10.0
 openshift==0.11.2
 passlib==1.7.4
-PyYAML==5.4.1
+#PyYAML==5.4.1
```
Comment (sampaccoud) [inline requirements/production.txt]: what not remove them completely?

Comment (jmaupetit) [reply requirements/production.txt]: This is a temporary test.

File: requirements/production.txt [hunk 2]
```diff
@@ -3,5 +3,4 @@ ansible==4.2.0
 jmespath==0.10.0
 openshift==0.11.2
```
Comment (jmaupetit) [inline requirements/production.txt]: Not sure we still need this dependency...

Comment (jmaupetit) [reply requirements/production.txt]: It seems to be required https://docs.ansible.com/ansible/latest/scenario_guides/kubernetes_scenarios/scenario_k8s_object.html

Comment (sampaccoud) [reply requirements/production.txt]: Ah 😏

File: tasks/set_vars.yml [hunk 3]
```diff
@@ -117,7 +117,7 @@
   set_fact:
     # We use escaped double quotes in the first regex_replace filter to be able
     # to add single quotes around app name in the query
-    apps_filter_query: "{{ apps_filter.split(',') | map('regex_replace', '(.*)', \"name=='\\1'\") | join(' || ') | regex_replace('(.*)', '[?\\1]') }}"
+    apps_filter_query: "[?{{ apps_filter.split(',') | map('regex_replace', '(\\w+)', \"name=='\\1'\") | join(' || ') }}]"
```
Comment (lunika) [inline tasks/set_vars.yml:120]: And what about all the `json_query` made to create the vaults ? They still work ?

**Comment by lunika** [comment]
Did you try this version locally by mounting it in matsuo instead of using the last release ?

**Comment by jmaupetit** [comment]
> Did you try this version locally by mounting it in matsuo instead of using the last release ?

Of course!

File: tasks/set_vars.yml (hunk 3 above)
Comment (jmaupetit) [reply tasks/set_vars.yml:120]: Yup.

**Review by lunika — APPROVED** [review]

**Pull request merged**